src/NCronJob/Registry/JobRun.cs

```diff
@@ -124,7 +124,7 @@ private void SetState(JobState state)
 
     public void NotifyStateChange(JobStateType type, Exception? fault = default)
     {
-        if (CurrentState.Type == type || IsCompleted)
+        if ((CurrentState.Type == type && type != JobStateType.Retrying) || IsCompleted)
```

Comment:
```suggestion
        var stateIsUnchangedOrCompleted = (CurrentState.Type == type && type != JobStateType.Retrying) || IsCompleted;
        if (stateIsUnchangedOrCompleted)
```

Reply:
Not applied as is, but reshaped in the same spirit. Better?